app/helpers/trends_helper.rb

```diff
@@ -33,8 +33,15 @@ def chart_data(trend, from = Setting.max_trend, to = Time.zone.now)
     values = trend.values
     labels = {}
     values.includes(:trendable).each {|v| labels[v.id] = [v.to_label, trend_path(:id => v)]}
-    values.includes(:trend_counters).where(["trend_counters.created_at > ?", from]).reorder("trend_counters.created_at").each_with_index.map do |value, idx|
-      data =  value.trend_counters.map { |t|  [t.created_at.to_i*1000, t.count]  }
+    values.includes(:trend_counters).where(["trend_counters.interval_start > ? or trend_counters.interval_end > ? or trend_counters.interval_end is null", from, from]).reorder("trend_counters.interval_start").each_with_index.map do |value, idx|
+      data = []
+      value.trend_counters.each do |counter|
+        #cut the left side of the graph
+        interval_start = counter.interval_start > from ? counter.interval_start : from
+        next_timestamp = counter.try(:interval_end) || Time.now
```

Comment:
Interval end can be `NULL`, if the interval has not been ended yet.
The timestamp is stored as `timestamp without time zone` (at least in pgsql), and rails puts there the UTC value as far as I have checked, so it should be OK with timezones.